### ISSUE TITLE
Minor problem with kindlepdfviewer-v2012.09-64-g6ea40b4.zip

### ISSUE THREAD

**Issue by dracodoc** [koreader/kindlepdfviewer#367]
The new release is great, now the crengine cache works correctly, the opening time for epub book is fast enough after the first time.

One minor problem I found is:
When pressed menu button for status, there will be a "retrieving TOC" message, but it will not appear after the first time. No matter if there is a TOC or not, this message always pop up in first time and no further updates. It's a little bit confusing, I'm not sure what this is about.

**Comment by tigran123** [comment]
@dracodoc Thank you, yes, we know about this, but it is not a bug really. When you press Menu to get the status you are going to press it again before you continue reading, right? And when you do that then the message will disappear, so nothing is obstructing your reading. Only while you are reading the status (for the first time) you can't see the text behind the "Retrieving TOC..." window.

I could easily fix this but it would mean adding an extra redraw of the page after the first press of Menu and I thought it is not necessary (i.e. it would just waste the battery).

But if you strongly believe that this is a bug, I can fix it in a minute. Think about this and let me know.

Oh btw, the reason why it is retrieving TOC (Table of Contents) is so that it can show which section the current page belongs to (at the bottom of the status). And, as you said, this will only happen for the first press of Menu --- the TOC is retrieved from the physical file only once (obviously the file cannot change :)

**Comment by dracodoc** [comment]
That's all right, I don't think it is a bug, just it confused me. With the message, I expect there will be some update when the retrieving is done, but there is no change and the message is kept on, which may lead user to think kpv is still retrieving TOC and wait some time to see the change, while in fact it probably already done that.

If I remember correctly, the previous version kpv can also show the current location to TOC in status, but there is no such message. Now with this message, what's the benefit?

So I'm thinking the message should just be removed, instead of show it then redraw the page.

**Comment by tigran123** [comment]
Ah, you see the reason the message must be shown on retrieving the physical TOC from disk is that for some files it can literally take _minutes_ and if there is no indication as to what is going on then the user will think that KPV is dead/locked up and will start resetting the Kindle.

On the other hand, soon you will be able to disable such messages altogether. And here is how:
1. In file manager press Alt-M and select "Expert mode"
2. In file manager press I and select "Messages"
3. Select option Q "Avoid any notifications".
4. Back to file manager, open the book and press Menu --- no "Retrieving TOC..." message is displayed.

Well, the above will work when the rest of @NuPogodi 's changes to new InfoMessage:inform() API are fully merged. The particular piece of code responsible for displaying "Retrieving TOC..." message has not been converted yet.

**Comment by tigran123** [comment]
Actually, I am thinking... maybe we should kill that message, as you suggest. This will simplify some other code which was specifically written to deal with this situation...

And as for those huge files with massive TOC --- well, two things:
1. they are very rare (the only ones I know of are my own editions of ancient texts morphologically parsed with TOC entries for each individual verse, i.e. each page contains only 2-3 verses and there are tens of thousands of pages)
2. the readers of such rare editions really ought to expect that the TOC manipulation is going to be slow and be patient...

Yes, let's kill it. I'll submit a PR and let others judge/decide.

**Comment by tigran123** [comment]
Done, see my PR #368

**Comment by dracodoc** [comment]
Just removing that message could be simpler. 

To make the message really tell user what kpv is doing, it at least need to update when there is no TOC or the retrieving has been finished, otherwise user always see the message before he press any key, but there are at least 3 possibilities with same message:
1. There is no TOC
2. There is a TOC, and the retrieving has been done after a while.
3. There is a TOC, kpv is retrieving it and still need more time to finish.

So user cannot tell which case it is from the message.

For massive TOC, I think maybe the status function should not try to retrieve all TOC if it took too long. Because the only way to get the location is for status function, and every time the book is closed then opened again, kpv need to retrieve TOC again, that means it will take a long time to just see the location.

**Comment by tigran123** [comment]
Fixed by the PR #368 so closing the issue.

PS. Btw, there is no way to predict if the TOC is going to be massive or not short of actually retrieving it.

**Comment by tigran123** [comment]
@dracodoc Try the latest build from:

https://github.com/tigran123/kindlepdfviewer/downloads

It will make you happy --- not only "Retrieving TOC..." is gone, but also writing crengine's warning/debug messages to Kindle's slow terminal (thousands of them!) has been disabled and so the handling of epub, fb2, etc ebook files is a bit faster.

**Comment by dracodoc** [comment]
Thanks, I tried this version, it worked great. All previous opened epub will be opened quickly, except in a new kpv session the first time of opening an epub may take several seconds, it looks like it's the time needed to load crengine, after that all is quite fast.
